test/system/035-logs.bats

```diff
@@ -298,4 +298,85 @@ $contentC" "logs -f on exitted container works"
 
     _log_test_follow journald
 }
+
+function _log_test_follow_since() {
+    local driver=$1
+    cname=$(random_string)
+    content=$(random_string)
+    local events_backend=$(_additional_events_backend $driver)
+
+
+    if [[ -n "${events_backend}" ]]; then
+        skip_if_remote "remote does not support --events-backend"
+    fi
+
+    run_podman ${events_backend} run --log-driver=$driver --name $cname $IMAGE sh -c "echo $content"
+    run_podman ${events_backend} logs --since 0s -f $cname
+    assert "$output" == "" "logs --since -f on exited container works"
+
+    run_podman ${events_backend} rm -t 0 -f $cname
+
+    # Now do the same with a running container to check #16950.
+    run_podman ${events_backend} run --log-driver=$driver --name $cname -d $IMAGE \
+        sh -c "while :; do echo $content && sleep 3; done"
+
+    # Make sure podman logs actually follows by giving a low timeout and check that the command times out
+    PODMAN_TIMEOUT=2 run_podman 124 ${events_backend} logs --since 1s -f $cname
+    assert "$output" == "$content
+timeout: sending signal TERM to command ‘$PODMAN’" "logs --since -f on running container works"
+
+    run_podman ${events_backend} rm -t 0 -f $cname
+}
+
+@test "podman logs - --since --follow k8s-file" {
+    _log_test_follow_since k8s-file
+}
+
+@test "podman logs - --since --follow journald" {
+    # We can't use journald on RHEL as rootless: rhbz#1895105
+    skip_if_journald_unavailable
+
+    _log_test_follow_since journald
+}
```

Comment:
These two tests pass on `main`.

Reply:
oh yeah you are right, I have to add a `sleep 1` before ` PODMAN_TIMEOUT=2 run_podman 124 ...` to make it fail. The logs command need to wait for the amount specified with --since in order to reproduce. I think I can try to use something like 0.5 seconds to make CI a bit faster.

Reply:
also this only effects the journald backend, the k8s-file should already work but I like to test both regardless